resources/services/shield/subscriptions.go

```diff
@@ -0,0 +1,147 @@
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/shield"
+	"github.com/aws/aws-sdk-go-v2/service/shield/types"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+//go:generate cq-gen --resource subscriptions --config gen.hcl --output .
+func Subscriptions() *schema.Table {
+	return &schema.Table{
+		Name:          "aws_shield_subscriptions",
+		Description:   "Information about the Shield Advanced subscription for an account",
+		Resolver:      fetchShieldSubscriptions,
+		Multiplex:     client.AccountMultiplex,
+		IgnoreError:   client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter:  client.DeleteAccountFilter,
+		Options:       schema.TableCreationOptions{PrimaryKeys: []string{"arn"}},
+		IgnoreInTests: true,
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSAccount,
+			},
+			{
+				Name:        "protection_group_limits_max_protection_groups",
+				Description: "The maximum number of protection groups that you can have at one time",
+				Type:        schema.TypeInt,
+				Resolver:    resolveSubscriptionsProtectionGroupLimitsMaxProtectionGroups,
+			},
+			{
+				Name:        "protection_group_limits_arbitrary_pattern_limits_max_members",
+				Description: "The maximum number of resources you can specify for a single arbitrary pattern in a protection group",
+				Type:        schema.TypeInt,
+				Resolver:    resolveSubscriptionsProtectionGroupLimitsArbitraryPatternLimitsMaxMembers,
+			},
+			{
+				Name:        "protected_resource_type_limits",
+				Description: "The maximum number of resource types that you can specify in a protection",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveSubscriptionsProtectedResourceTypeLimits,
+			},
+			{
+				Name:        "auto_renew",
+				Description: "If ENABLED, the subscription will be automatically renewed at the end of the existing subscription period",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "end_time",
+				Description: "The date and time your subscription will end",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "limits",
+				Description: "Specifies how many protections of a given type you can create",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveSubscriptionsLimits,
+			},
+			{
+				Name:        "proactive_engagement_status",
+				Description: "If ENABLED, the Shield Response Team (SRT) will use email and phone to notify contacts about escalations to the SRT and to initiate proactive customer support",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "start_time",
+				Description: "The start time of the subscription, in Unix time in seconds",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "arn",
```

Comment:
Same here do subscriptions have tags?

Reply:
it seems that only protections and protection groups have tags
https://aws.amazon.com/about-aws/whats-new/2021/03/aws-shield-advanced-now-supports-resource-tagging/